crates/oxc_linter/src/rules/eslint/no_loss_of_precision.rs

```diff
@@ -0,0 +1,302 @@
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct NormalizedNum {
+    magnitude: isize,
+    coefficient: String,
+}
+
+impl NoLossOfPrecision {
+    fn get_raw(node: &'_ NumberLiteral) -> String {
+        node.raw.replace('_', "")
+    }
+
+    fn base_ten(node: &'_ NumberLiteral) -> bool {
+        let prefixes = ["0x", "0X", "0b", "0B", "0o", "0O"];
+        !prefixes.iter().any(|prefix| node.raw.starts_with(prefix))
+    }
+
+    fn not_base_ten_loses_precision(node: &'_ NumberLiteral) -> bool {
+        let raw = Self::get_raw(node).to_uppercase();
+        #[allow(clippy::cast_possible_truncation, clippy::cast_sign_loss)]
+        // AST always store number as f64, need a cast to format in bin/oct/hex
+        let value = node.value as u64;
+        let suffix = if raw.starts_with("0B") {
+            format!("{value:b}")
+        } else if raw.starts_with("0X") {
+            format!("{value:x}")
+        } else {
+            format!("{value:o}")
+        };
+        !raw.ends_with(&suffix.to_uppercase())
+    }
+
+    fn base_ten_loses_precision(node: &'_ NumberLiteral) -> bool {
+        let normalized_raw_num = if let Some(s) = Self::normalize(&Self::get_raw(node)) {
+            s
+        } else { return true };
+        let precision = normalized_raw_num.split('e').next().unwrap().replace('.', "").len();
+
+        if precision > 100 {
+            return true;
+        }
+
+        let stored_num = format!("{:1$}", node.value, precision);
+        let normalized_stored_num = if let Some(s) = Self::normalize(&stored_num) {
+            s
+        } else {return true};
+        normalized_raw_num != normalized_stored_num
+    }
+
+    fn add_decimal_point(num: &str) -> String {
+        format!("{}.{}", &num[..1], &num[1..])
+    }
+
+    fn normalize_int(num: &str) -> NormalizedNum {
+        // specially deal with 0
+        if num == "0" {
+            return NormalizedNum { magnitude: 0, coefficient: String::from("0.") };
+        }
+
+        let significant_digits = num.trim_start_matches('0').trim_end_matches('0');
+        #[allow(clippy::cast_possible_wrap)]
+        // the length of number is larger then isize is almost impossible in real-world codebase
+        let magnitude =
+            if num.starts_with('0') { num.len() as isize - 2 } else { num.len() as isize - 1 };
+        NormalizedNum { magnitude, coefficient: Self::add_decimal_point(significant_digits) }
+    }
+
+    fn normalize_float(num: &str) -> NormalizedNum {
+        let trimmed_float = match num.trim_start_matches('0') {
+            "" => "0",
+            s => s,
+        };
+
+        return trimmed_float.strip_prefix('.').map_or_else(
+            || {
+                let trimmed_float = trimmed_float.trim_end_matches('0');
+                // unwrap here will never panic, we guarantee the input contains a `.`
+                #[allow(clippy::cast_possible_wrap)]
+                let magnitude = (trimmed_float.find('.').unwrap() - 1) as isize;
```

Comment:
Try getting rid of the `unwrap`.

Reply:
Not sure if it is necessary to remove the `unwrap` here, since we have already checked the existence of `.` at [here](https://github.com/web-infra-dev/oxc/pull/649/files/3ad40f3b10e04eb89c0959c79774d64eefe51a11#diff-93d98222d2f2dd962f9301ba43aa2a1035ffd22a1a0beca80ccae1e26cb02167R157), so I personally think the `unwrap` here is ok, but if it's for code style purposes, I'm glad to fix it 😄